kano_updater/commands/download.py

```diff
@@ -76,21 +78,34 @@ def download(progress=None, gui=True):
         RCState.get_instance().rc = RC.NO_NETWORK
         return False
 
+    priority = Priority.NONE
+
+    if status.is_urgent:
+        priority = Priority.URGENT
+
     if not is_server_available():
         err_msg = N_("Could not connect to the download server")
         logger.error(err_msg)
         progress.fail(_(err_msg))
         RCState.get_instance().rc = RC.CANNOT_REACH_KANO
         return False
 
+    run_cmd('sudo kano-empty-trash')
+    enough_space, space_msg = check_disk_space(priority)
```

Comment:
Not necessarily a problem but the`install()` function calls the `download()` function and expects a `True` return to continue with its things - this means that the `install()` will end up checking for disk space twice within a few seconds of one another.

Reply:
That's true. Let's do the check twice during an install for now and see how that performs. Can't hurt tbh.